vmdb/app/models/ems_kubernetes.rb

```diff
@@ -13,17 +13,41 @@ def self.description
 
   def self.raw_connect(hostname, port, api_version)
     require 'kubeclient'
-    require 'uri'
-
-    uri = URI::HTTP.build(:path => "/api", :port => port.to_i)
+    api_endpoint = raw_api_endpoint(hostname, port)
+    Kubeclient::Client.new(api_endpoint, api_version)
+  end
 
+  def self.raw_api_endpoint(hostname, port)
+    require 'uri'
     # URI::Generic#hostname= was added in ruby 1.9.3 and will automatically
     # wrap an ipv6 address in []
-    uri.hostname = hostname
-    Kubeclient::Client.new uri.to_s, api_version
+    URI::HTTP.build(:host => hostname, :port => port.to_i, :path => '/api')
```

Comment:
Is it possible that it's unrelated to my refactoring? I don't think I changed anything in that regard and if it's a separate improvement maybe doesn't belong to my patch.